src/net_processing.cpp

```diff
@@ -2535,8 +2535,9 @@ bool ProcessMessage(CNode* pfrom, const std::string& msg_type, CDataStream& vRec
 
     if (msg_type == NetMsgType::TX) {
         // Stop processing the transaction early if
-        // We are in blocks only mode and peer is either not whitelisted or whitelistrelay is off
-        // or if this peer is supposed to be a block-relay-only peer
+        // 1) We are in blocks only mode and peer is either not whitelisted or the peer is whitelisted
```

Comment:
```suggestion
        // 1) We are in blocks only mode and the peer has no relay permission
```

I think this can be said slightly more concise, but still correct